node-js-client/test/local/reputation_service_client_tests.js

```diff
@@ -0,0 +1,119 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+var test = require('tap').test
+var IPReputationClient = require('../../lib/client')
+var client = new IPReputationClient({
+  host: '127.0.0.1',
+  port: 8080,
+  id: 'root',
+  key: 'toor'
+})
+
+test(
+  'throws exception when missing required config values',
+  function (t) {
+    t.throws(function () { new IPReputationClient({}) })
+    t.throws(function () { new IPReputationClient({port: 8080, id: 'root', key: 'toor'}) })
+    t.throws(function () { new IPReputationClient({host: '127.0.0.1', id: 'root', key: 'toor'}) })
+    t.throws(function () { new IPReputationClient({host: '127.0.0.1', port: 8080, key: 'toor'}) })
+    t.throws(function () { new IPReputationClient({host: '127.0.0.1', port: 8080, id: 'root'}) })
+    t.end()
+  }
+)
+
+test(
+  'does not get reputation for a nonexistent IP',
+  function (t) {
+    client.get('127.0.0.1', function (error, response, body) {
+      t.equal(error, null)
+      t.equal(response.statusCode, 404)
+      t.end()
+    })
+  }
+)
+
+test(
+  'does not update reputation for nonexistent IP',
+  function (t) {
+    client.update('127.0.0.1', 5, function (error, response, body) {
+      t.equal(error, null)
+      t.equal(response.statusCode, 200)
```

Comment:
Is it correct that the service returns `200` if the item does not exist and can not be updated?